data_steward/cleaners/null_invalid_foreign_keys.py

```diff
@@ -0,0 +1,116 @@
+"""
+
+ Cleaning Rule 1 : Foreign key references (i.e. visit_occurrence_id in the condition table) should be valid.
+ (Existing Achilles rule - validating for foreign keys include provider_id, care_site_id, location_id, person_id,
+ visit_occurrence_id)
+
+
+"""
+
+# Project Imports
+import resources
+from constants.cleaners.clean_cdr import PERSON_TABLE_NAME
+
+FOREIGN_KEYS_FIELDS = ['person_id', 'visit_occurrence_id', 'location_id', 'care_site_id', 'provider_id']
+
+INVALID_FOREIGN_KEY_QUERY = '''
+            SELECT {cols} 
+            FROM {project}.{dataset_id}.{table_name} t 
+            {join_expr}
+            '''
+
+
+def mapping_table_for(domain_table):
+    """
+    Get name of mapping table generated for a domain table
+
+    :param domain_table: one of the domain tables (e.g. 'visit_occurrence', 'condition_occurrence')
+    :return: mapping table name
+    """
+    return '_mapping_' + domain_table
+
+
+def null_invalid_foreign_keys(project_id, dataset_id):
```

Comment:
This generates a list of SELECT queries.  When is it actually updating the value to null?

Reply:
I am using this strategy because, we can simply achieve that using the Left join to the tables where those keys are primary keys.

Reply:
You're right, the left join will return all the records from the left table and will fill null values for any record in the right table.  However, I thought that BigQuery created temporary result sets.  SELECT statements won't actually update the table values.   Or have I missed something in the BQ documentation?

Reply:
the query function in bq_utils will update the original table if we pass, destination table and destination dataset_id.

Reply:
Then that change needs to be made as part of this PR.  But it can't break the existing queries.